dts/bindings/video/video-interfaces.yaml

```
@@ -0,0 +1,133 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+# Common properties for video interface endpoints.
+
+include: base.yaml
+
+compatible: "zephyr,video-interfaces"
+
+description: |
+  A video pipeline usually consists of several devices, e.g. camera sensors, video
+  data receivers, video data processors, etc. Data interfaces on these devices are
+  described by their child 'port' nodes. Configuration of a port depends on other
+  devices in the pipeline and is described by 'endpoint' subnodes.
+
+  If a port can be configured to work with more than one remote device on the same
+  bus, an 'endpoint' child node must be provided for each of them. If more than one
+  port is present in a device node or there is more than one endpoint at a port, or
+  port node needs to be associated with a selected hardware interface, a common
+  scheme using '#address-cells', '#size-cells' and 'reg' properties is used.
+
```

How about showing examples for each case? The text is not ambiguous and easy to read, but having an example could waive the last doubts.

```suggestion

    video_device {
      ...
      port {
          ...
          endpoint@0 { ... };
          endpoint@1 { ... };
      };
    };
```

Most video video devices could also be in this situation rather than the multi-port multi-endpoint situation.

---

Oh, I see video devices with multiple ports are more common on single port and multiple endpoints :). For example, sensor interfaces (receiver) devices such as mipi csi2rx are multiple ports.
About the examples for '#address-cells', '#size-cells' and 'reg', I will try to put some examples for this. Thanks